### DISCUSSION TITLE
We need to do some serious refactoring

### DISCUSSION THREAD

**Discussion by squivix** [Benji377/OpenMusic#78] [Ideas]
I think it's about time we clean up the code a little and agree on some style guides for naming our resources. We need to come up with a consistent way for naming:

layout resource files
id's
string resources
colors

We could follow a [popular convention](https://stackoverflow.com/questions/7165582/are-there-conventions-on-how-to-name-resources) or agree on our own.

We also need to use those colors and strings everywhere because later when we want to add themes and translations it's gonna be real useful to have.

**Comment by Benji377** [comment]
I read some answers on StackOverflow and I quite like this answer: https://stackoverflow.com/a/39866695/14159914
But I'm still a little bit unsure. I would also like to rename classes and packages. "com.example.blabla" seems so unprofessional to me, but I have no idea to what to rename it, "com.music.blabla" maybe?
Let me know what you think about it and if you found some better conventions you prefer.

PS: Yeah you are totally right, the colors thing is super messed up because I at the start didn't understand which color Android chose (sometimes it colored some stuff purple) so I just changed every color tag I could find to red, just to be sure.

**Comment by squivix** [comment]
I liked that answer too. It's actually what got me to click on that link. We could publish the cheat sheet in the contribute section and start refactoring. Is it settled then?

As for the class and package names I'm a bit less sure. I don't know good practices but I know the com.example.app thing is supposed to be the domain name then company name the app name.

**Reply by Benji377** [reply]
Oh right and since we aren't a company it doesn't matter. Ok then. And yep, if you are ok with it it's settled. I will add it to the contributing section tomorrow.

**Reply by squivix** [reply]
Ok done with strings, layout files, and id's in pull request #85.

Only colors left. I will have to read up on this because I'm not too familiar with themes.

EDIT: never mind. I just noticed you already changed colors. Good job :D
